Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/PurposeOfForDecider.java

```diff
@@ -60,11 +59,10 @@ public boolean isPurposeFor(CallbackProperties properties) {
         boolean purposeFor = false;
 
         String action = properties.getAction();
-        try {
-            serviceName = NHIN_SERVICE_NAMES.fromValueString(action);
-        } catch (IllegalArgumentException ex) {
-            LOG.warn("Could not read purpose of / for action: {}", ex.getLocalizedMessage());
-            LOG.trace("Could not read purpose of / for action: {}", ex.getLocalizedMessage(), ex);
+
+        serviceName = NHIN_SERVICE_NAMES.fromValueString(action);
+        if (null == serviceName) {
+            LOG.warn("Could not read purpose of / for action: service name is null");
```

Comment:
Can we include the service name that was not found for troubleshooting later on?